### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by BrenoDeMedeiros**
Adding a descriptive documentation in the README

Signed-off-by: BrenoDeMedeiros <breno@google.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,2 @@
 # authservice
-Prototype of external auth service
+An implementation of [Envoy](https://envoyproxy.io) [External Authorization](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/ext_authz_filter), focused on delivering authN/Z for [Kubernetes](https://kubernetes.io) and [Istio](https://istio.io).
```
Comment (liminw) [inline README.md]: Do we want to mention Kubernetes? Istio supports many different platforms.

Comment (BrenoDeMedeiros) [reply README.md]: Well, some of the authentication mechanisms we intend to develop are specific to the Kubernetes control plane. That doesn't mean that the Istio-oriented solutions we develop are specific to Kubernetes.

What about if instead I said "authN/Z for Kubernetes and Istio" I said "authN/Z solutions for Cloud Native platforms, such as Istio and Kubernetes" ?

Comment (liminw) [reply README.md]: Sounds good!

Comment (BrenoDeMedeiros) [reply README.md]: I simplified it a bit to 'solutions for Istio and Kubernetes'. I think inverting the order is sufficient, while adding a reference to Cloud Native could cause confusion given that Istio also manages non-Cloud Native workloads; also some might associate Cloud Native to CNCF and Istio is not a project there. Let me know what you think.

Comment (liminw) [reply README.md]: That's a good point. I agree that mentioning CNCF here is causing more confusion. I am fine with the current description.

**Pull request merged**